sqlglot/dialects/tsql.py

```diff
@@ -983,7 +983,10 @@ def setitem_sql(self, expression: exp.SetItem) -> str:
             return super().setitem_sql(expression)
 
         def boolean_sql(self, expression: exp.Boolean) -> str:
-            if type(expression.parent) in BIT_TYPES:
+            if (
+                type(expression.parent) in BIT_TYPES
+                or expression.find_ancestor(exp.Values) is not None
```

Comment:
You can make the search a bit more efficient by stopping at the first `Select`:
```suggestion
                or isinstance(expression.find_ancestor(exp.Values, exp.Select), exp.Values)
```